src/node/splitid_endpoints.h

```diff
@@ -0,0 +1,643 @@
+
+          rss->remove(in.session_id);
+          return make_success();
+        }
+        catch (std::exception& ex)
+        {
+          return make_error(
+            HTTP_STATUS_BAD_REQUEST, ccf::errors::InternalError, ex.what());
+        }
+      };
+
+    endpoints
+      ->make_endpoint(
+        "splitid/update-identity",
+        HTTP_POST,
+        json_adapter(splitid_update_identity),
+        {std::make_shared<NodeCertAuthnPolicy>()})
+      .set_forwarding_required(endpoints::ForwardingRequired::Always)
+      .install();
+
+    auto get_identity = [&context](auto& args, const nlohmann::json& params) {
+      try
+      {
+        auto ctx = get_splitid_ctx(context);
+
+        if (!ctx)
+        {
+          return make_error(
+            HTTP_STATUS_NOT_FOUND,
+            ccf::errors::InternalError,
+            "No split identity context");
+        }
+
+        auto rss = args.tx.rw(ctx->resharing_sessions);
+        auto sss = args.tx.rw(ctx->sampling_sessions);
+        if (rss->size() != 0 || sss->size() != 0)
+        {
+          return make_error(
+            HTTP_STATUS_SERVICE_UNAVAILABLE,
+            ccf::errors::InternalError,
+            "A sampling or resharing session is in progress");
+        }
+
+        CurrentIdRPC::Out out;
+        auto cid = args.tx.ro(ctx->current_identity);
+        auto r = cid->get(0);
+        if (!r.has_value())
+        {
+          return make_error(
+            HTTP_STATUS_SERVICE_UNAVAILABLE,
+            ccf::errors::ResourceNotFound,
+            "no identity available yet");
+        }
+        else
+        {
+          out.pem = EC::Point(r.value().public_key).to_public_pem();
+          return make_success(out);
+        }
+      }
+      catch (std::exception& ex)
+      {
+        return make_error(
+          HTTP_STATUS_BAD_REQUEST, ccf::errors::InternalError, ex.what());
+      }
+    };
+
+    endpoints
+      ->make_endpoint(
+        "splitid/get-identity",
+        HTTP_POST,
+        json_adapter(get_identity),
+        no_auth_required)
+      .install();
+
+#define SPLITID_INTERNAL_ENDPOINT(URL, T, ST, TBL, F) \
+  auto T##_endpoint_fun = \
+    [&context](auto& args, const nlohmann::json& params) { \
+      try \
+      { \
+        const auto in = params.get<T::In>(); \
+        SessionTx<ST> stx(args, in.session_id, TBL); \
+        stx.session.F(stx.from, in.payload); \
+        stx.put(); \
+        return make_success(true); \
+      } \
+      catch (std::exception & ex) \
+      { \
+        return make_error( \
+          HTTP_STATUS_BAD_REQUEST, ccf::errors::InternalError, ex.what()); \
+      } \
+    }; \
+  endpoints \
+    ->make_endpoint( \
+      URL, \
+      HTTP_POST, \
+      json_adapter(T##_endpoint_fun), \
+      {std::make_shared<NodeCertAuthnPolicy>()}) \
+    .set_forwarding_required(endpoints::ForwardingRequired::Always) \
+    .set_openapi_hidden(true) \
```

Comment:
If I'm reading correctly, then replacing this line with `.set_auto_schema<T::In, bool>()` should add useful auto-generated OpenAPI entries for these endpoints (at least saying "the request should be a `SamplingDealRPC::In`).